src/message_bus.zig

```diff
@@ -188,18 +191,15 @@ fn MessageBusType(comptime process_type: vsr.ProcessType) type {
             allocator.free(bus.replicas_connect_attempts);
         }
 
-        fn init_tcp(io: *IO, address: std.net.Address) !struct {
-            fd: IO.socket_t,
-            address: std.net.Address,
-        } {
+        fn init_tcp(io: *IO, family: u32) !IO.socket_t {
             const fd = try io.open_socket(
-                address.any.family,
+                family,
                 posix.SOCK.STREAM,
                 posix.IPPROTO.TCP,
             );
             errdefer io.close_socket(fd);
 
-            const address_resolved = try io.listen(fd, address, .{
+            try io.socket_options(fd, .{
```

Comment:
Instead of separate `io.socket_options` call, consider adding `options` parameter to `io.open_socket` so that, from the perspective of the caller, we atomically get the `fd` we want.

Reply:
We can't do that because some options are TCP-specific, and we use `io.open_socket` for UDP connections.
Renamed `socket_options` to `tcp_options` instead, so we can add `udp_options` when/if required.

Reply:
Hm, maybe we can `open_socket_tcp` / `open_socket_udp`, so that we also don't have to pass `posix.IPPROTO.TCP` which feels like it's at the wrong abstraction layer here?